mzLib/Proteomics/ProteolyticDigestion/PeptideWithSetModifications.cs

```diff
@@ -1250,7 +1251,222 @@ public PeptideWithSetModifications GetReverseDecoyFromTarget(int[] revisedAminoA
             }
 
         }
+        /// <summary>
+        /// This function generates a decoy peptide from a target by scrambling the target peptide's amino acid sequence
+        /// This preserves any digestion motifs and keeps modifications with their amino acids
+        /// To help generate only high quality decoys, a homology cutoff of 30 % sequence similarity is used
+        /// If after 10 attempts no sufficient decoy is generated, the mirror sequence is returned
+        /// </summary>
+        /// <param name="revisedAminoAcidOrder">Array to store the new amino acid order in</param>
+        /// <param name="maximumHomology">Parameter specifying the homology cutoff to be used</param>
+        /// <returns></returns>
+        public PeptideWithSetModifications GetScrambledDecoyFromTarget(int[] revisedAminoAcidOrder, double maximumHomology = 0.3)
+        {
```

Comment:
Why is revisedAminoAcidOrder a parameter? I feel like it would be simpler to initialize an empty array inside the method. Do you anticipate the revisedAminoAcidOrder array being useful outside of testing?

Reply:
I kept it as a parameter because it was a parameter in the original which generated decoys by just reversing the sequence.

Reply:
Sounds good